src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -266,7 +285,20 @@ def _forward(self, model_inputs):
 
     def postprocess(self, model_outputs):
         if self.type == "ctc_with_lm":
-            logits = np.concatenate([outputs["logits"].numpy() for outputs in model_outputs], axis=1)
+            final_logits = []
+            for outputs in model_outputs:
+                logits = outputs["logits"].numpy()
+                stride = outputs.get("stride", None)
+                if stride is not None:
+                    total_n, left, right = stride
+                    # Total_n might be < logits.shape[1]
+                    # because of padding, that's why
+                    # we need to reconstruct this information
+                    # This won't work with left padding (which doesn't exist right now)
+                    right_n = total_n - right
+                    logits = logits[:, left:right_n]
```

Comment:
Nice, yes I think that's the only approach that'll work right now

Reply:
Padding is not possible here sadly

Reply:
Padding works, it's handled by the batching mecanism. All I mentionned here is why we need so much information (`logits` might get padded while `stride` no).

We *can* make it work, relatively trivially for left padding.

```python
if self.tokenizer.padding_side == "left":
    left = logits.shape[1] - total_n + left_n
    right = logits.shaoe[1] - total_n + right_n
else:
   left = left_n
   right_n = total_n - right_n
```
Just thought this was overly complex since left padding doesn't seem likely here.

Reply:
Ah I see - ok yeah